libraries/AP_HAL/system.cpp

```diff
@@ -0,0 +1,6 @@
+#include "system.h"
+
+uint16_t WEAK AP_HAL::millis16()
+{
+    return millis64() & 0xFFFF;
```

Comment:
No particular reason - just assuming the base type to be 64-bit.

Will change to millis().  

This will make it less efficient on NuttX, but we're beyond caring there.